lib/puppet/provider/package/gem.rb

```diff
@@ -108,6 +108,15 @@ def install(useversion = true)
     self.fail "Could not install: #{output.chomp}" if output.include?("ERROR")
   end
 
+  # Allow Gem version expressions in ensure
+  def insync?(is)
+    begin
+      Gem::Dependency.new('', resource[:ensure]).match?('', is)
+    rescue
+      false
```

Comment:
This method is not a standard location for the ensure parameter, so it is not called by default. It is only called by `insync?` in the type when all other options have failed. This is why this works with present/absent.